graphql/handler/apollofederatedtracingv1/tree_builder.go

```diff
@@ -11,12 +12,14 @@ import (
 
 	"github.com/99designs/gqlgen/graphql"
 	"github.com/99designs/gqlgen/graphql/handler/apollofederatedtracingv1/generated"
+	"github.com/vektah/gqlparser/v2/gqlerror"
 )
 
 type TreeBuilder struct {
-	Trace    *generated.Trace
-	rootNode generated.Trace_Node
-	nodes    map[string]NodeMap // nodes is used to store a pointer map using the node path (e.g. todo[0].id) to itself as well as it's parent
+	Trace        *generated.Trace
+	rootNode     generated.Trace_Node
+	nodes        map[string]NodeMap // nodes is used to store a pointer map using the node path (e.g. todo[0].id) to itself as well as it's parent
+	errorOptions TraceErrors
```

Comment:
```suggestion
	errorOptions *TraceErrors
```
It looks like this `TraceErrors` struct is passed by value, but when you modify, I think you always want to replace the subfields of the original version, so you should just change it to a pointer to avoid the possible bug.